masonry/src/widget/flex.rs

```diff
@@ -852,18 +852,11 @@ impl Widget for Flex {
             major = total_major;
         }
 
+        // my_size may be larger than the given constraints.
+        // In which case, the Flex widget will either overflow its parent
+        // or be clipped (e.g. if its parent is a Portal).
         let my_size: Size = self.direction.pack(major, minor_dim).into();
 
-        // if we don't have to fill the main axis, we loosen that axis before constraining
-        let my_size = if !self.fill_major_axis {
-            let max_major = self.direction.major(bc.max());
-            self.direction
-                .constraints(bc, 0.0, max_major)
-                .constrain(my_size)
-        } else {
-            bc.constrain(my_size)
-        };
-
```

Comment:
Upon re-read, I'm realizing I'm not sure what this block does (specifically the fill_major_axis part), so removing it entirely may be ill-advised.

Reply:
Upom re-re-read, I think it's fine. This block essentially applies both minimum constraints and maximum constraints to the returned size. We specifically don't want to apply maximum constraints (so scrolling works), and minimum constraints are applied inconsistently in Masonry anyway.